File: 14_Batch1_EducationalInstitutes/C++/educationalInstitutesOfIndia.cpp
// Author:Ethan D'Costa
//  Roll No:13
//  MCA
//  Start Date:24-7-24
#include <iostream>
#include <string>
using namespace std;

class Edu_Instititutes
{
public:
    string Sno;
    string InstituteName;
};

class InstitutesofNatlImportance : public Edu_Instititutes
{
public:
    string City, State, Act_name, Admin_dept;

    InstitutesofNatlImportance(string sno, string instituteName, string city, string state, string act_name, string admin_dept)
        : Edu_Instititutes{sno, instituteName}, City{city}, State{state}, Act_name{act_name}, Admin_dept{admin_dept} {}

    void printInstitutes()
    {
        cout << "Information about Institutes of National Importance:\n";
        cout << "Sno: " << Sno << "\n";
        cout << "InstituteName: " << InstituteName << "\n";
        cout << "City: " << City << "\n";
        cout << "State: " << State << "\n";
        cout << "Act Name: " << Act_name << "\n";
        cout << "Dept: " << Admin_dept << "\n";
        cout << "-------------------------------------------\n";
    }
};

class UniversitiesofIndia : public Edu_Instititutes
{
public:
    string UniName, State, Year;

    UniversitiesofIndia(string sno, string uniName, string state, string year)
        : Edu_Instititutes{sno, uniName}, State{state}, Year{year} {}

    void printInstitutes()
    {
        cout << "Information about Universities of India:\n";
        cout << "Sno: " << Sno << "\n";
        cout << "InstituteName: " << InstituteName << "\n";
        cout << "State: " << State << "\n";
        cout << "Year:" << Year << "\n";
        cout << "-------------------------------------------\n";
    }
};

int main()
{
    UniversitiesofIndia u1("1", "Aligarh Muslim University", "Uttar Pradesh", "1964");
    u1.printInstitutes();

    InstitutesofNatlImportance im1("1", "Aligarh Muslim University", "Aligarh", "Uttar Pradesh",
                                   "'Entry No. 63, Union list - The 7th schedule under Article 246 of the Constitution of India'",
                                   "Ministry of Education");
    im1.printInstitutes();

    return 0;
}

File: 14_Batch1_EducationalInstitutes/C#/educationalInstitutesOfIndia.cs
//Author:Ethan D'Costa
// Roll No:13
// MCA
// Start Date:24-7-24
using System;
namespace EducationalInstitutes
{
      class EduInstitutes
      {
            public string sno, InstituteName;
      }
      class InstitutesOfNatlImportance : EduInstitutes
      {
            public string City, State, ActName, AdminDept;
            public InstitutesOfNatlImportance(string sno, string instituteName, string city, string state, string actName, string adminDept)
            {
                  this.sno = sno;
                  this.InstituteName = instituteName;
                  this.City = city;
                  this.State = state;
                  this.ActName = actName;
                  this.AdminDept = adminDept;
            }

            public void DisplayInstitutes()
            {
                  Console.WriteLine("Information about Institutes of National Importance:");
                  Console.WriteLine($"Sno: {sno}");
                  Console.WriteLine($"InstituteName: {InstituteName}");
                  Console.WriteLine($"City: {City}");
                  Console.WriteLine($"State: {State}");
                  Console.WriteLine($"Act Name: {ActName}");
                  Console.WriteLine($"Dept: {AdminDept}");
                  Console.WriteLine("-------------------------------------------");
            }

      }
      class UniversitiesOfIndia : EduInstitutes
      {
            public string State, Year;
            public UniversitiesOfIndia(string sno, string uniName, string state, string year)
            {
                  this.sno = sno;
                  this.InstituteName = uniName;
                  this.State = state;
                  this.Year = year;
            }

            public void DisplayInstitutes()
            {
                  Console.WriteLine("Information about Universities of India:");
                  Console.WriteLine($"Sno: {sno}");
                  Console.WriteLine($"InstituteName: {InstituteName}");
                  Console.WriteLine($"State: {State}");
                  Console.WriteLine($"Year: {Year}");
                  Console.WriteLine("-------------------------------------------");
            }
      }

      class EduInstititutesofIndia
      {
            static void Main(string[] args)
            {
                  InstitutesOfNatlImportance inst1 = new InstitutesOfNatlImportance("1", "Atal Bihari Vajpayee Indian Institute of Information Technology and Management,Gwalior", "Gwalior", "Madhya Pradesh", "The Indian Institutes of Information Technology Act 2014 and their subsequent amendments", "Ministry of Education");
                  UniversitiesOfIndia u1 = new UniversitiesOfIndia("1", "Aligarh Muslim University", "Uttar Pradesh", "1964");
                  inst1.DisplayInstitutes();
                  u1.DisplayInstitutes();
            }
      }


}